xsamtools/cram.py

```diff
@@ -125,6 +125,13 @@ def encode_itf8(value):
         raise ValueError('Number is too large for an unsigned 32-bit integer.')
     return bytes(integers)
 
+def decode_itf8_array(handle, size=None):
+    number_of_items_in_array = decode_itf8(handle) if size is None else size
+    itf8_array = []
+    for _ in range(number_of_items_in_array):
+        itf8_array.append(decode_itf8(handle))
+    return itf8_array
```

Comment:
Instead of introducing a new function and corresponding tests, consider creating arrays on the fly:
```
foo = [decode_itf8(fh) for _ in range(size)]
```

Reply:
Size is still a part of the data structure generally so I'd like to have a function that just knows that and does it for you.

Reply:
I can add the comprehension as the return value if that makes things clearer.

Reply:
This should be done and ready for a re-review.